packages/adapter/src/lib/adapter/adapter.ts

```diff
@@ -168,117 +1088,594 @@ class AdapterClass extends EventEmitter {
+
+        /**
+         * Promise-version of Adapter.subscribeForeignStates
+         */
+        this.subscribeForeignStatesAsync = tools.promisify(this.subscribeForeignStates, this);
+
+        /**
+         * Promise-version of Adapter.unsubscribeForeignStates
+         */
+        this.unsubscribeForeignStatesAsync = tools.promisify(this.unsubscribeForeignStates, this);
+
+        /**
+         * Promise-version of Adapter.subscribeStates
+         */
+        this.subscribeStatesAsync = tools.promisify(this.subscribeStates, this);
+
+        /**
+         * Promise-version of Adapter.unsubscribeStates
+         */
+        this.unsubscribeStatesAsync = tools.promisify(this.unsubscribeStates, this);
+
+        /**
+         * Promise-version of Adapter.setBinaryState
+         *
+         * @param {string} id of state
+         * @param {Buffer} binary data
+         * @param {object} [options] optional
+         * @return {Promise}
+         *
+         */
+        this.setForeignBinaryStateAsync = tools.promisify(this.setForeignBinaryState, this);
+
+        /**
+         * Async version of setBinaryState
+         *
+         *
+         * @param {string} id of state
+         * @param {Buffer} binary data
+         * @param {object} [options] optional
+         * @param {Promise<void>}
+         */
+        this.setBinaryStateAsync = tools.promisify(this.setBinaryState, this);
+
+        /**
+         * Promise-version of Adapter.getBinaryState
+         *
+         *
+         */
+        this.getForeignBinaryStateAsync = tools.promisify(this.getForeignBinaryState, this);
+
+        /**
+         * Promisified version of getBinaryState
+         *
+         * @param {string} id The state ID
+         * @param {object} options optional
+         * @return {Promise<Buffer>}
+         */
+        this.getBinaryStateAsync = tools.promisify(this.getBinaryState, this);
+
+        /**
+         * Promise-version of Adapter.delForeignBinaryState
+         *
+         * @param {string} id
+         * @param {object} [options]
+         * @return {Promise<void>}
+         *
+         */
+        this.delForeignBinaryStateAsync = tools.promisify(this.delForeignBinaryState, this);
+
+        /**
+         * Promise-version of Adapter.delBinaryState
+         *
+         * @param {string} id
+         * @param {object} [options]
+         * @return {Promise<void>}
+         *
+         */
+        this.delBinaryStateAsync = tools.promisify(this.delBinaryState, this);
+
+        this.setExecutableCapabilities = tools.setExecutableCapabilities;
+        this._init();
+    }
+
+    // overload with real types
+    decrypt(secretVal: string, value?: string): string;
+    /**
+     * Decrypt the password/value with given key
+     * @param secretVal to use for decrypt (or value if only one parameter is given)
+     * @param  [value] value to decrypt (if secret is provided)
+     */
+    decrypt(secretVal: unknown, value?: unknown): string {
```

Comment:
I like having all public signatures annotated with `public`, to make it very obvious that these are meant to be public methods.
```suggestion
    // overload with real types
    public decrypt(secretVal: string, value?: string): string;
    /**
     * Decrypt the password/value with given key
     * @param secretVal to use for decrypt (or value if only one parameter is given)
     * @param  [value] value to decrypt (if secret is provided)
     */
    public decrypt(secretVal: unknown, value?: unknown): string {
```

Reply:
I mark every private method as private, so public is default as it of course is in TS. I still get your intention, we have not done this until now so I am quite unsure if we should change for this PR.

Reply:
We can do that later. I think I started doing it because some recommended ESLint rule enforces/enforced it.

Reply:
Then there is probably an autofix for the whole codebase. Then fine for me to add this rule later on.